pkg/drivers/hyperkit/driver.go

```diff
@@ -447,17 +448,24 @@ func (port InvalidPortNumberError) Error() string {
 }
 
 func (d *Driver) extractVSockPorts() ([]int, error) {
-	vsockPorts := make([]int, 0, len(d.VSockPorts))
+	convertedPorts := func(yield func(int) bool) {
+		for _, port := range d.VSockPorts {
+			if p, err := strconv.Atoi(port); err == nil {
+				if !yield(p) {
+					return
+				}
+			}
+		}
+	}
 
+	// Check for errors first
 	for _, port := range d.VSockPorts {
-		p, err := strconv.Atoi(port)
-		if err != nil {
+		if _, err := strconv.Atoi(port); err != nil {
 			return nil, InvalidPortNumberError(port)
 		}
-		vsockPorts = append(vsockPorts, p)
 	}
 
-	return vsockPorts, nil
+	return slices.Collect(convertedPorts), nil
```

Comment:
Previous code was better, let's not use slices.Collect with yield, only with existing iterators like maps.Keys().